R/summary_bound.R

```diff
@@ -0,0 +1,231 @@
+#'                             hr = c(1, .6), 
+#'                             dropoutRate = .001)
+#' # Information fraction
+#' IF <- (1:3)/3 
+#' # Analysis times in months; first 2 will be ignored as IF will not be achieved
+#' analysisTimes <- c(.01, .02, 36)  
+#' # Experimental / Control randomization ratio
+#' ratio <- 1 
+#' # 1-sided Type I error
+#' alpha <- 0.025 
+#' # Type II error (1 - power)
+#' beta <- .1 
+#' # Upper bound
+#' upper <- gsdmvn::gs_spending_bound
+#' upar <- list(sf = gsDesign::sfLDOF, total_spend = 0.025, param = NULL, timing = NULL)
+#' # Lower bound
+#' lower <- gsdmvn::gs_spending_bound
+#' lpar <- list(sf = gsDesign::sfHSD, total_spend = 0.1, param = 0, timing = NULL)
+#' # weight function in WLR
+#' wgt00 <- function(x, arm0, arm1){
+#'   gsdmvn:::wlr_weight_fh(x, arm0, arm1, rho = 0, gamma = 0)}
+#' wgt05 <- function(x, arm0, arm1){
+#'   gsdmvn:::wlr_weight_fh(x, arm0, arm1, rho = 0, gamma = .5)}
+#' # test in COMBO
+#' fh_test <- rbind(
+#'   data.frame(rho = 0, gamma = 0, tau = -1, test = 1, Analysis = 1:3,analysisTimes = c(12, 24, 36)), 
+#'   data.frame(rho = c(0, 0.5), gamma = 0.5, tau = -1, test = 2:3, Analysis = 3, analysisTimes = 36)
+#' )
+#' # ---------------------------- #
+#' #          ahr                 #
+#' # ---------------------------- #
+#' x_ahr <- gs_design_ahr2(
+#'   enrollRates = enrollRates,
+#'   failRates = failRates,
+#'   IF = IF, # Information fraction
+#'   analysisTimes = analysisTimes, 
+#'   ratio = ratio, 
+#'   alpha = alpha, 
+#'   beta = beta, 
+#'   upper = upper,
+#'   upar = upar,
+#'   lower = lower,
+#'   lpar = lpar
+#'   )
+#' summary_bound(x_ahr, method = "AHR")
+#' # ---------------------------- #
+#' #         wlr                  #
+#' # ---------------------------- #
+#' x_wlr2 <- gs_design_wlr2(
+#'   enrollRates = enrollRates,
+#'   failRates = failRates,
+#'   weight = wgt05, 
+#'   IF = NULL, 
+#'   analysisTimes = sort(unique(x_ahr$analysis$Time)), 
+#'   ratio = ratio, 
+#'   alpha = alpha, 
+#'   beta = beta,   
+#'   upper = upper,
+#'   upar = upar,
+#'   lower = lower,
+#'   lpar = lpar
+#'   )
+#'   summary_bound(x_wlr, method = "WLR")
+#' # ---------------------------- #
+#' #         max combo            #
+#' # ---------------------------- #
+#' x_combo <- gsdmvn::gs_design_combo2(
+#'   ratio = 1, 
+#'   alpha = 0.025, 
+#'   beta = 0.2, 
+#'   enrollRates = tibble::tibble(Stratum = "All", duration = 12, rate = 500/12),
+#'   failRates = tibble::tibble(Stratum = "All", duration = c(4, 100), 
+#'                              failRate = log(2) / 15, hr = c(1, .6), dropoutRate = .001), 
+#'   fh_test = fh_test,
+#'   upper = gsdmvn::gs_spending_combo,
+#'   upar = list(sf = gsDesign::sfLDOF, total_spend = 0.025),
+#'   lower = gsdmvn::gs_spending_combo,
+#'   lpar = list(sf = gsDesign::sfLDOF, total_spend = 0.2))
+#' summary_bound(x_combo, method = "COMBO")
+#' 
+summary_bound <- function(
+  x,
+  method = NULL,
```

Comment:
If `method` is a required argument and has a definitive number of possible values, then explicitly list them. https://design.tidyverse.org/def-enum.html

Making it `NULL` here indicates it is optional and can be `NULL`, which is not the case.